src/runner.ts

```diff
@@ -35,11 +36,11 @@ async function loadMigrations(
 
     const migrations = await Promise.all(
       files.map(async (file) => {
-        const filePath = `${options.dir}/${file}`;
+        const filePath = resolve(options.dir, file);
         const actions: MigrationBuilderActions =
           extname(filePath) === '.sql'
             ? await migrateSqlFile(filePath)
-            : require(relative(__dirname, filePath));
+            : createRequire(resolve('noop.js'))(filePath);
```

Comment:
**question (non-blocking):** is this a leftover? or did you not rebase on the latest `main`-branch? 👀  
If this is still in the `main`-branch this indeed needs to be fixed.  
Could you extract this out into its own PR please?

Reply:
This branch is up to date.

I can extract this in a separate PR, but removing the last commit should be enough. Or not?

Reply:
Yes, we now use this (or #1207) as the PR that will fix the esm source code issue

I really appreciate your help and I'm learning new things, but as I wrote in my other comment regarding the squash-merge PRs, try to make dedicated PRs